### PR TITLE
Fix defaults that were set to empty string in collections.

### PR DESCRIPTION
**Pull request by sean-m-sullivan**
##### SUMMARY
Fix several places where an empty string was set as default instead of nothing. This is related to #9200 . Currently if user does not set the field, but it was set in tower, it would overwrite the value with an empty string. With this change the original value stays unless specified. 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - awx-collection

##### AWX VERSION
17.0.1

### CODE REVIEW

File: awx_collection/plugins/modules/tower_ad_hoc_command.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def main():
         limit=dict(),
         credential=dict(required=True),
         module_name=dict(required=True),
-        module_args=dict(default=""),
+        module_args=dict(),
```
Comment (john-westcott-redhat) [inline awx_collection/plugins/modules/tower_ad_hoc_command.py:121]: Can you confirm there is a test on this one? Down on like 148 module_args is used to build a post_data without a conditional `if module_args`

File: awx_collection/plugins/modules/tower_project.py [hunk 2]
```diff
@@ -229,8 +229,8 @@ def main():
         scm_type=dict(choices=['manual', 'git', 'svn', 'insights'], default='manual'),
         scm_url=dict(),
         local_path=dict(),
-        scm_branch=dict(default=''),
-        scm_refspec=dict(default=''),
+        scm_branch=dict(),
+        scm_refspec=dict(),
```
Comment (john-westcott-redhat) [inline awx_collection/plugins/modules/tower_project.py:233]: Also make sure there are tests for these as well. Lines 322/323 are hard coding these into the project_files unconditionally.

**Review by john-westcott-redhat — CHANGES_REQUESTED** [review]

File: awx_collection/plugins/modules/tower_project.py (hunk 2 above)
Comment (sean-m-sullivan) [reply awx_collection/plugins/modules/tower_project.py:233]: Checked this one, and we normally don't reference them, so looks like tests passed. https://github.com/ansible/awx/blob/4e4811870418b17ec173bd7651b675003abb572a/awx_collection/tests/integration/targets/tower_project/tasks/main.yml#L78

File: awx_collection/plugins/modules/tower_ad_hoc_command.py (hunk 1 above)
Comment (sean-m-sullivan) [reply awx_collection/plugins/modules/tower_ad_hoc_command.py:121]: I'm working on this one, have to make a test for it.

Comment (sean-m-sullivan) [reply awx_collection/plugins/modules/tower_ad_hoc_command.py:121]: Test added, does not look to affect the module.

**Review by beeankha — APPROVED** [review]

**Pull request merged**